src/Donations/Models/Donation.php

```diff
@@ -190,11 +192,11 @@ public function getSequentialId()
     /**
      * @since 2.19.6
      *
-     * @return object[]
+     * @return ModelQueryBuilder
      */
-    public function getNotes()
+    public function notes()
```

Comment:
Seems like it would be useful to also have a `$donation->addNote($message)` method which creates a note and appends the note to the donation's `cachedRelations['notes']` **if** its set.

Reply:
fix: https://github.com/impress-org/givewp/compare/d8f860cad587...de4c08a2ce5c https://github.com/impress-org/givewp/compare/410cf6573cee...48d811ee84a3

Reply:
@JasonTheAdams `$cachedRelations` is a private property of the parent class. I did not find a method to overwrite/edit this property.
Can you suggest?

Reply:
@JasonTheAdams One way to resolve this is to make `$cachedRelations` property `protected`. Then edit it from `addNote` function but I did not like the way we have to manually edit this property in the Model class.

Reply:
@JasonTheAdams @ravinderk I made the decision to revert this abstraction for now in favor of using the `DonationNote` model directly.  I really want to think through the right approach for creating model relationship instances before introducing helpers like this that could likely end up being deprecated or changed down the line.  Ideally we'll have a more consistent api for this scenario that all models can use.  ref: f517b364

Reply:
I'm fine with that. It's certainly a reliable way of going about it. We can think more about helpers later.